File: src/main/java/org/dandoy/dbpop/Table.java
package org.dandoy.dbpop;

import java.util.List;

class Table {
    private final TableName tableName;
    private final List<Column> columns;
    private final List<Index> indexes;
    private final List<ForeignKey> foreignKeys;

    Table(TableName tableName, List<Column> columns, List<Index> indexes, List<ForeignKey> foreignKeys) {
        this.tableName = tableName;
        this.columns = columns;
        this.indexes = indexes;
        this.foreignKeys = foreignKeys;
    }

    @Override
    public String toString() {
        return "Table{" +
                "tableName=" + tableName +
                '}';
    }

    TableName getTableName() {
        return tableName;
    }

    List<Column> getColumns() {
        return columns;
    }

    List<Index> getIndexes() {
        return indexes;
    }

    List<ForeignKey> getForeignKeys() {
        return foreignKeys;
    }
}

File: dbpopr/src/models/Table.ts
import {TableName} from "./TableName";
import {PrimaryKey} from "./PrimaryKey";
import {ForeignKey} from "./ForeignKey";

export interface Column {
    name: string;
    nullable: boolean;
    autoIncrement: boolean;
}

export interface Index {
    name: string;
    tableName: TableName;
    unique: boolean;
    primaryKey: PrimaryKey | null;
    columns: string[];
}

export interface Table {
    tableName: TableName;
    columns: Column[];
    indexes: Index[];
    primaryKey: PrimaryKey | null;
    foreignKeys: ForeignKey[];
}